### PR TITLE
boot_from_virtiofs: Update console for different arch

### PR DESCRIPTION
**Pull request by SiboWang1997**
Different arch use different console, need to update kernel_params on different arch.

ID: 2215182
Signed-off-by: Sibo Wang siwang@redhat.com

### CODE REVIEW

**Comment by SiboWang1997** [comment]
JOB ID     : a3ab48cc1bbfddeee54021af5e9916ab27a72ad3
JOB LOG    : /root/avocado/job-results/job-2023-07-23T23.10-a3ab48c/job.log
 (1/1) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.boot_from_virtiofs.arm64-pci: STARTED
 (1/1) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.boot_from_virtiofs.arm64-pci: PASS (166.19 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-07-23T23.10-a3ab48c/results.html
JOB TIME   : 167.34 s

**Comment by SiboWang1997** [comment]
Hello @YongxueHong @zhenyzha @xiagao , could you help me review this patch when you are free? Thanks a lot.

File: qemu/tests/cfg/boot_from_virtiofs.cfg [hunk 1]
```diff
@@ -27,5 +27,8 @@
     mem-path_mem0 = /dev/shm
     backend_mem_mem0 = memory-backend-file
     share_mem = yes
-    kernel_params = "root=virtiofs:${fs_target} console=ttyS0"
+    x86_64:
+        kernel_params = "root=virtiofs:${fs_target} console=ttyS0"
+    aarch64:
+        kernel_params = "root=virtiofs:${fs_target} console=ttyAMA0"
```
Comment (YongxueHong) [inline qemu/tests/cfg/boot_from_virtiofs.cfg:33]:
```suggestion
    kernel_params = "root=virtiofs:${fs_target} console=ttyS0"
    aarch64:
        kernel_params = "root=virtiofs:${fs_target} console=ttyAMA0"
```

**Review by YongxueHong — DISMISSED** [review]

**Comment by zhenyzha** [comment]
git am 3782.patch
Applying: boot_from_virtiofs: Add aarch64 console "ttyAMA0"

```
 (1/2) Host_RHEL.m9.u3.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.boot_from_virtiofs.arm64-pci: PASS (167.53 s)
 (2/2) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.boot_from_virtiofs.arm64-pci: PASS (163.51 s)
```

LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Review by zhenyzha — APPROVED** [review]

File: qemu/tests/cfg/boot_from_virtiofs.cfg [hunk 2]
```diff
@@ -28,4 +28,6 @@
     backend_mem_mem0 = memory-backend-file
     share_mem = yes
     kernel_params = "root=virtiofs:${fs_target} console=ttyS0"
+    aarch64:
+        kernel_params = "root=virtiofs:${fs_target} console=ttyAMA0"
```
Comment (PaulYuuu) [inline qemu/tests/cfg/boot_from_virtiofs.cfg:32]: We should consider all architectures, in my opinion:
```suggestion
    x86_64, i386:
        console_dev = "ttyS0,115200"
    ppc64le, ppc64:
        console_dev = "hvc0,38400"
    aarch64:
        console_dev = "ttyAMA0,38400"
    s390x:
        console_dev = "ttysclp0,38400"
    kernel_params = "root=virtiofs:${fs_target} console=${console_dev}"
        
```

**Review by PaulYuuu — CHANGES_REQUESTED** [review]

**Comment by SiboWang1997** [comment]
Hi @fbq815 @MiriamDeng ,could you please help review this patch when you are free? Thanks a lot.

**Comment by MiriamDeng** [comment]
> Add console for aarch64.
> 
> ID: 2215182 Signed-off-by: Sibo Wang [siwang@redhat.com](mailto:siwang@redhat.com)

Here, we should remove aarch64 here.

**Comment by MiriamDeng** [comment]
ACK

**Review by PaulYuuu — APPROVED** [review]

**Comment by YongxueHong** [comment]
Hi @SiboWang1997 
For the CI checking failure, please rebase(update) your development branch to the latest.
Thanks.

**Comment by SiboWang1997** [comment]
> Hi @SiboWang1997 For the CI checking failure, please rebase(update) your development branch to the latest. Thanks.

The CI checking failure is caused by #3786 , should I update the branch when #3786 merged?

**Review by PaulYuuu — APPROVED** [review]

**Comment by PaulYuuu** [comment]
@SiboWang1997 Now you can rebase your branch to fix the CI issue.

**Review by YongxueHong — APPROVED** [review]

**Comment by PaulYuuu** [comment]
Thanks all.

**Pull request merged**